src/Features/Core/Portable/ConflictMarkerResolution/AbstractConflictMarkerCodeFixProvider.cs

```diff
@@ -67,105 +62,119 @@ protected AbstractResolveConflictMarkerCodeFixProvider(string diagnosticId)
-                                ? FeaturesResources.Take_top
-                                : string.Format(FeaturesResources.Take_0, topText);
-
-                            var bottomText = endTrivia.ToString().Substring(s_mergeConflictLength).Trim();
-                            var takeBottomText = string.IsNullOrWhiteSpace(bottomText)
-                                ? FeaturesResources.Take_bottom
-                                : string.Format(FeaturesResources.Take_0, bottomText);
-
-                            var startSpan = startTrivia.Span;
-                            var equalsSpan = equalsTrivia.Span;
-                            var endSpan = endTrivia.Span;
-                            
-                            context.RegisterCodeFix(
-                                new MyCodeAction(takeTopText, 
-                                    c => TakeTopAsync(document, startSpan, equalsSpan, endSpan, c)),
-                                context.Diagnostics);
-                            context.RegisterCodeFix(
-                                new MyCodeAction(takeBottomText, 
-                                c => TakeBottomAsync(document, startSpan, equalsSpan, endSpan, c)),
-                                context.Diagnostics);
-                            context.RegisterCodeFix(
-                                new MyCodeAction(FeaturesResources.Take_both, 
-                                c => TakeBothAsync(document, startSpan, equalsSpan, endSpan, c)),
-                                context.Diagnostics);
+                            RegisterCodeFixes(context, startTrivia, equalsTrivia, endTrivia);
+                            return;
                         }
                     }
+
+                    if (index + 2 < token.LeadingTrivia.Count)
+                    {
+                        // case where there is ===== followed by >>>>>>  on the next line.
+
+                        var equalsTrivia = leadingTrivia[index];
+                        var endOfLineTrivia = leadingTrivia[index + 1];
+                        var endTrivia = leadingTrivia[index + 2];
+
+                        if (IsEndOfLine(endOfLineTrivia) &&
+                            IsConflictMarker(text, endTrivia, '>'))
+                        {
+                            RegisterCodeFixes(context, startTrivia, equalsTrivia, endTrivia);
+                            return;
+                        }
+                    }
+                }
+
+                token = token.GetNextToken(includeZeroWidth: true);
+                if (token.RawKind == 0)
+                {
+                    return;
                 }
             }
         }
 
+        private void RegisterCodeFixes(
+            CodeFixContext context, SyntaxTrivia startTrivia, SyntaxTrivia equalsTrivia, SyntaxTrivia endTrivia)
+        {
+            var document = context.Document;
+
+            var topText = startTrivia.ToString().Substring(s_mergeConflictLength).Trim();
```

Comment:
I know it was already like this, but would it be better to `.Trim()` before `.Substring(...)`? That way if it starts with whitespace and then has real text, you get the real text instead of trimming away all of the whitespace leading substring...